platform/admin/admin-configuration-configupdater/src/main/java/org/codice/ddf/admin/configuration/ConfigurationUpdater.java

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.configuration;
+
+import static com.google.common.io.Files.getFileExtension;
+import static java.lang.String.format;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.security.common.audit.SecurityLogger;
+import ddf.security.encryption.EncryptionService;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.admin.core.api.ConfigurationAdmin;
+import org.codice.ddf.platform.io.internal.PersistenceStrategy;
+import org.codice.felix.cm.internal.ConfigurationContext;
+import org.codice.felix.cm.internal.ConfigurationContextFactory;
+import org.codice.felix.cm.internal.ConfigurationPersistencePlugin;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.metatype.AttributeDefinition;
+import org.osgi.service.metatype.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Addresses shortcomings of the Felix {@code ConfigInstaller} with our own implementation that
+ * properly updates the etc directory.
+ *
+ * <p>This configuration installer handles <b>writeback</b> like the Felix one did and does a few
+ * other things:
+ *
+ * <ul>
```

Comment:
Perhaps a note regarding the `enableConfigSave` value in `custom.properties` in case integrators decide to flip the switch while still running this bundle.